src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/ISMTemplateService.kt

```diff
@@ -98,30 +139,368 @@ fun validateFormat(indexPatterns: List<String>): ElasticsearchException? {
     return null
 }
 
+fun List<ISMTemplate>.findSelfConflictingTemplates(): Pair<List<String>, List<String>>? {
+    val priorityList = this.map { it.priority }
+    priorityList.forEach { priority ->
+        // same priority
+        val indexPatternsList = this.filter { it.priority == priority }.map { it.indexPatterns }
```

Comment:
If I'm not mistaken, if we're trying to check the indexPatterns of any templates with the same priority, it might be more efficient to create a Map of `priority -> List<ISMTemplate>` first so that way you can iterate through the priorities and then only check the indexPatterns when the list size is greater than 1 for a particular priority key in the Map.

This will help avoid doing `this.filter` each time just to get matching priorities.

Reply:
sure, changed